text/0002-new-architecture.md

```diff
@@ -0,0 +1,243 @@
+* Start Date: 2022-07-21
+* RFC Type: informational
+* RFC PR: https://github.com/getsentry/rfcs/pull/2
+
+# Summary
+
+This document is a living piece that is updated with some collected thoughts on what likely projects
+are over a multi year horizon to improve Sentry's underpinnings.
+
+# Motivation
+
+We are running into scaling limitations on our current infrastructure and as such some larger
+decisions have to be made about the future of our codebase.  As we are talking about creating
+more Sentry deployments in different regions and for smaller clusters, some of these problems
+are becoming more pressing as they create complexity for each added installation.
+
+# Expected Workstreams
```

Comment:
I think we may have another work stream to deal with.
We have a few independent code bases today (a few independent services) and likely, in the future, we will have more. Though we do not have a Sentry Foundation set of services and libraries that solve common problems for all the code bases services, thus we end up rebuilding things.
There is a number of common components I am thinking about:
- rate limiting libraries (a number of rate limiters policies. Today we literally reimplemented that in Snuba and Sentry)
- A runtime config flags service+sdk with admin UI usable by everybody. We have 3 systems now, none of which is good.
- Shared settings system across services.
- Traffic routing sdk.
- A streaming platform that allows product teams to run their product logic without caring too much about the deployment of a Kafka consumer. If we move more towards Kafka to replace rabbit we will have a lot more consumers running product logic.
- Postgres models (we will certainly not allow different services to share a postgres instance, though it does not mean that different services could access their own postgres instance and take advantage of reusing the sentry abstractions)
- ....